src/gentropy/dataset/variant_index.py

```diff
@@ -25,6 +26,30 @@
 class VariantIndex(Dataset):
     """Dataset for representing variants and methods applied on them."""
 
+    def __post_init__(self: VariantIndex) -> None:
```

Comment:
This is fine for the time being, but I think it is not elegant to have ad hoc code here for a situation that is also true for other datasets.
If you agree, I propose creating a separate issue to avoid iterables being represented with nulls (and have an empty array for example). This will ensure we don't fall into such errors when doing array operations

Reply:
To be honest I'm happy with this code and considering removing this function. I mean applying this post init function will not prevent nulls if there's a join operation on the datasets, which is exactly the case why I added this in the first place.

Reply:
It can be added to the df setter method.
Let's leave it for now